src/python/cli_new/lib/cli/config.py

```diff
@@ -119,6 +120,79 @@ def master(self):
 
         return master
 
+    def principal(self):
+        """
+        Return the principal in the configuration file
+        """
+        return self.data["master"].get("principal")
+
+    def secret(self):
+        """
+        Return the secret in the configuration file
+        """
+        return self.data["master"].get("secret")
+
+    def agent_ssl(self, default=False):
+        """
+        Return if the agent support ssl
+        """
+        if "agent" in self.data:
+            agent_ssl = self.data["agent"].get("ssl", default)
+            if not isinstance(agent_ssl, bool):
+                raise CLIException("The 'agent->ssl' field"
+                                   " must be True/False")
+
+            return agent_ssl
+
+        return default
```

Comment:
So you can simplify it even further by using a default for `self.data["agent"], like this]:
```
def agent_ssl(self, default=False):
    agent_ssl = self.data.get("agent", {}).get("ssl", default)
    if not isinstance(agent_ssl, bool):
        raise CLIException("The 'agent->ssl' field must be True/False"
    return agent_ssl
```

Personally I like it because it removes a repetition between checking that `"agent"` is in `data` and then accessing it, and also it removes a branch, but up to you if you find it less natural.

Reply:
I also was thinking to replace it like that, but in my eyes it looks too complicated. :thinking: You don't think so?

Reply:
Up to you, I'm fine either way :).